internal/rbd/controllerserver.go

```diff
@@ -294,27 +259,123 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	return &csi.CreateVolumeResponse{Volume: volume}, nil
 }
 
+func (cs *ControllerServer) checkFlatten(ctx context.Context, rbdVol *rbdVolume, cr *util.Credentials) error {
```

Comment:
should this not be a method of `rbdVolume` instead? `cs` is not used in this function, only `rbdVolume`.

Reply:
earlier i stored the hard and soft depth flags in cs. after that i removed. fixed now